orderbook/src/api/get_fee_and_price.rs

```diff
@@ -0,0 +1,186 @@
+use super::H160Wrapper;
+use crate::fee::{MinFeeCalculating, MinFeeCalculationError};
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+use ethcontract::{H160, U256};
+use model::{order::OrderKind, u256_decimal};
+use num::BigInt;
+use serde::{Deserialize, Serialize};
+use shared::{
+    conversions::U256Ext,
+    price_estimate::{PriceEstimating, PriceEstimationError},
+};
+use std::convert::Infallible;
+use std::sync::Arc;
+use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
+
+#[derive(Deserialize)]
+struct Query {
+    sell_token: H160Wrapper,
+    buy_token: H160Wrapper,
+    #[serde(with = "u256_decimal")]
+    amount: U256,
+    kind: OrderKind,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Response {
```

Comment:
I liked about the markets route that it tells you which token the amounts refer to (to make confusing the API's response harder). Not sure this is needed for the fee (which is always sell token) but I do think it would be nice for the amount.

Reply:
For this route I think it makes more sense to not specify the token as the amount is always expressed in the other token with respect to the input amount. Adding the token only introduces ambiguity.
 I have added comments to the struct explain this.